sdk/src/types/block/output/delegation.rs

```diff
@@ -355,6 +357,53 @@ impl DelegationOutput {
             .locked_address(self.address(), context.essence.creation_slot())
             .unlock(unlock, inputs, context)
     }
+
+    // Transition, just without full ValidationContext.
+    pub(crate) fn transition_inner(current_state: &Self, next_state: &Self) -> Result<(), StateTransitionError> {
+        if !(current_state.delegation_id.is_null() && !next_state.delegation_id().is_null()) {
+            return Err(StateTransitionError::NonDelayedClaimingTransition);
+        }
```

Comment:
This code should never be reached, since the next state is determined by the id, no?

Reply:
It's more to prevent transitions from explicit ID to explicit ID as this doesn't seem to be allowed